src/NServiceBus.Transport.Msmq/MsmqTransport.cs

```diff
@@ -32,7 +32,7 @@ public override TransportInfrastructure Initialize(SettingsHolder settings, stri
             Guard.AgainstNull(nameof(settings), settings);
             string errorQueue;
 
-            if (!settings.TryGetExplicitlyConfiguredErrorQueueAddress(out errorQueue))
+            if (!settings.TryGetExplicitlyConfiguredErrorQueueAddress(out errorQueue) && !settings.Get<bool>("Endpoint.SendOnly"))
```

Comment:
Wouldn't it be better to check for the SendOnly configuration condition first, and if and only then check the queue configuration? In other words, I think instead of appending to the condition, you should check for the send only endpoint first.